transport/src/test/java/io/netty/channel/DefaultChannelIdTest.java

```diff
@@ -84,4 +107,27 @@ public void testSerialization() throws Exception {
         assertThat(a, is(not(sameInstance(b))));
         assertThat(a.asLongText(), is(b.asLongText()));
     }
+
+    @Test
+    public void testDeserialization() throws Exception {
+        // deseiralize DefaultChannelId with 8 byte machineId
+        final ObjectInputStream in8 = new ObjectInputStream(new ByteArrayInputStream(EIGHT_BYTES_MACHINE_ID));
+        final DefaultChannelId c8 = (DefaultChannelId) in8.readObject();
```

Comment:
I'd prefer to see a `@VisibleForTesting` package-private constructor where you can pass in either the values or a finished byte array. Keeping the binaries around is troublesome for maintenance and debugging.

Reply:
Agree, Let me implement it :D

Reply:
Done. Could you please take a look?